src/value/borrowed.rs

```diff
@@ -348,22 +373,11 @@ impl<'val> Struct for BorrowedStruct<'val> {
 
 impl<'val> PartialEq for BorrowedStruct<'val> {
     fn eq(&self, other: &Self) -> bool {
-        // check if both the text_fields have same (field_name,value) pairs
-        self.text_fields.iter().all(|(key, value)| {
-            value.iter().all(|(_my_s, my_v)| {
-                other
-                    .get_all(key)
-                    .find(|other_v| my_v == *other_v)
-                    .is_some()
-            })
-        }) && self.no_text_fields.iter().all(|(my_k, my_v)| {
-            // check if both the no_text_fields have same values
-            other
-                .no_text_fields
-                .iter()
-                .find(|(other_k, other_v)| my_k == other_k && my_v == other_v)
-                .is_some()
-        })
+        // check if both test_fields and no_text_fields have same length
+        self.text_fields.len() == other.text_fields.len() && self.no_text_fields.len() == other.no_text_fields.len()
```

Comment:
Our current definition of `SymbolToken` defines that its text must be resolvable via `SymbolToken::text()` if possible (note that this does not imply that an implementation need to be eager/lazy).  So based on your statement above, I would expect (1) to be true.

We do have a definition in `SymbolToken` that this should be consistent with:

https://github.com/amzn/ion-rust/blob/4d0c67b3273bd62603ddc2a3c75d711348b0e4b8/src/value/mod.rs#L223-L259

A more friendly rendering is here:

https://docs.rs/ion-rs/0.5.0/ion_rs/value/trait.SymbolToken.html

Reply:
>The two structs being compared having the same encodings for each field, which would be too strict.

We cannot use `PartialEq` or `Eq` to compare across two different implementations, but I am not sure to what you mean with _same encodings_ for the field.  Do you mind clarifying this statement?

Reply:
Per our offline discussion: the current `SymbolToken`'s trait had some intentional ambiguity and we're going to revise some of the language.

>>    The two structs being compared having the same encodings for each field, which would be too strict.
>
> We cannot use PartialEq or Eq to compare across two different implementations, but I am not sure to what you mean with same encodings for the field. Do you mind clarifying this statement?

We established offline that `BorrowedSymbolToken` is required to be fully resolved (i.e. `text()` and `local_sid()` will be populated if at all possible), which renders this point moot.

Because `SymbolToken` did not have a hard requirement that its implementations provide resolved data, it was feasible that `BorrowedSymbolToken` could be used for both resolved and 'raw' use cases. If a `BorrowedStruct` was maintaining a collection of raw `BorrowedSymbolToken`s, then the line of code in question would be requiring that the two structs use the same field encodings. (`{foo: true}` and `{$12: true}` would not be considered equal even if `$12=foo`.)